src/Symfony/Component/Validator/Constraints/AutoMapping.php

```diff
@@ -0,0 +1,58 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Validator\Constraints;
+
+use Symfony\Component\Validator\Constraint;
+use Symfony\Component\Validator\Exception\ConstraintDefinitionException;
+
+/**
+ * Enables or disables auto mapping.
+ *
+ * If the constraint is present on:
+ *   * a property, and set to true, auto-mapping for this property is always enabled, regardless of the config, and of any class level annotation
+ *   * a property, and set to false, auto-mapping for this property is always disabled, regardless of the config, and of any class level annotation
+ *   * a class, and set to true, auto-mapping for this class is always enabled except if a the annotation has been added to a specific property, and regardless of the config
+ *   * a class, and set to false, auto-mapping is always disabled except if a the annotation has been added to a specific property, and regardless of the config
+ *
+ * @Annotation
+ *
+ * @author Kévin Dunglas <dunglas@gmail.com>
+ */
+class AutoMapping extends Constraint
```

Comment:
What about having two dedicated annotations (e.g. `EnableAutoMapping` and `DisableAutoMapping`) instead of the option switch?

Reply:
Your call!